students/EricAdams/session07/test_html_render.py

```diff
@@ -155,11 +155,77 @@ def test_render_non_strings():
     assert '<body>' in contents
     assert '</body' in contents
 
-    # we want the tesxt, too:
+    # we want the text, too:
     assert 'any string I like' in contents
 
     # now lets get pretty specific:
     # the opening tag should come before the ending tag
     assert contents.index('<body>') < contents.index('</body>')
     # the opening tag should come before the content
     assert contents.index('<body>') < contents.index('any string')
+
+
+def test_render_with_indent():
+    # this is crating a html page with a single body() element in it
+    el_object = Html(Body(P('any string I like')))
+
+    contents = render_element(el_object)
+    # make sure extra whitespace at beginning or end doesn't mess things up.
+    contents = contents.strip()
+
+    print(contents)  # so we can see what's going on if a test fails
+
+    # so what should the results be?
+    # the html tag is the outer tag, so the contents should start
+    # and end with that.
+    assert contents.startswith('<html>')
+    assert contents.endswith('</html>')
+
+    # the body tags had better be there too
```

Comment:
looking good!

though you may want to set the number of spaces somewhere in your test, and then use it. maybe even:

the_indent = Element.indent